File: nature-harvest-backend/nature-harvest-backend/src/main/java/com/api/nature_harvest_backend/dtos/blog/BlogDto.java
package com.api.nature_harvest_backend.dtos.blog;

import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class BlogDto {
    @NotNull(message = "Title is required")
    private String title;

    @NotNull(message = "Content is required")
    private String content;

    @NotNull(message = "Thumbnail is required")
    private String thumbnail;

    @NotNull(message = "Tags is required")
    private String tags;

    @NotNull(message = "Summary is required")
    private String summary;

}

File: nature-harvest-frontend/src/app/dtos/blog/blog.dto.ts
export class BlogDto {
  title: string;
  content: string;
  thumbnail: string;
  tags: string;
  summary: string;

  constructor(data: any) {
    this.title = data.title;
    this.content = data.content;
    this.thumbnail = data.thumbnail;
    this.tags = data.tags;
    this.summary = data.summary;
  }
}
